dsl/filter.go

```diff
@@ -0,0 +1,25 @@
+package dsl
+
+import (
+	"github.com/quasilyte/go-ruleguard/dsl/types"
+)
+
+// VarFilterContext carries Var and environment information into the filter function.
+// It's an input parameter type for the Var.Filter function callback.
+type VarFilterContext struct {
+	// Type is mapped to Var.Type field.
+	Type types.Type
+}
+
+// SizeOf returns the size of the given type.
+func (*VarFilterContext) SizeOf(x types.Type) int { return 0 }
```

Comment:
Maybe return `int64`, same as https://golang.org/pkg/go/types/#StdSizes.Sizeof.
This could happen for a large array.

Reply:
> Maybe return int64, same as https://golang.org/pkg/go/types/#StdSizes.Sizeof.

I need to think about this. Initially, I thought to support only `int` numerical type (and maybe `float64` and `byte` later) to simplify the compilation and interpretation. Assuming that most platforms are 64-bit, `int` would be as good as `int64`.

It's not that hard to implement all (or most) of the numerical types and make sure it works on both 64-bit and 32-bit platforms, but it's also very tempting to simplify some details to keep the implementation complexity sane.

> Also, I've noticed the StdSizes.Sizeof spelling vs SizeOf here.

I used `SizeOf` spelling in `TypeExpr` method. I wonder if we should use `Sizeof` in both places.